pandas/io/parsers.py

```diff
@@ -3607,8 +3607,8 @@ def __init__(self, f, colspecs, delimiter, comment, skiprows=None, infer_nrows=1
 
         if not isinstance(self.colspecs, (tuple, list)):
             raise TypeError(
-                "column specifications must be a list or tuple, "
-                "input was a %r" % type(colspecs).__name__
+                f"column specifications must be a list or tuple, "
```

Comment:
```suggestion
                "column specifications must be a list or tuple, "
```

The `f` prefix is only needed if actually substituting

Reply:
Noted. I'll change it tonight!